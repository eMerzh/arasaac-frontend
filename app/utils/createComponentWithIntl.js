import React from 'react'
import renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'
import { render } from 'enzyme'

const createComponentWithIntl = (children, props = { locale: 'en' }) =>
  renderer.create(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  )


export const intlEnzymeComponent = (children, props = { locale: 'en' }) =>
  render(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  )

export default createComponentWithIntl
