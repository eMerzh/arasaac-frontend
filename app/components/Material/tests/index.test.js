import React from 'react'
import createComponentWithIntl from 'utils/createComponentWithIntl'
import Component from '../index'

describe('<Material />', () => {
  it('renders correctly', () => {
    const material = {
      title: 'Title test',
      desc: 'Description for test material'
    }
    const tree = createComponentWithIntl(
      <Component material={material} locale={'es'} />)
    expect(tree).toMatchSnapshot()
  })
})
