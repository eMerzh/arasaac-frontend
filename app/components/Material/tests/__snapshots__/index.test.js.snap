exports[`<Material /> renders correctly 1`] = `
<li>
  <section>
    <p>
      es
    </p>
    <h1
      className="gufdDH">
      Title test
    </h1>
    <p>
      Description for test material
    </p>
  </section>
</li>
`;
